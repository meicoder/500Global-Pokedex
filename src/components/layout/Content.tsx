import { PokemonList } from '../pokemon/PokemonList';
import styled from 'styled-components';
import { useState } from 'react';
import { PokemonCard } from '../pokemon/PokemonCard';

const RowStyled = styled.div`
    background-color: #f40552;
    border: 10px solid #f40552;
    border-radius: 12px;
`;

const ColStyled = styled.div`
    background-color: rgba(252, 126, 47, 0.6);
    border: 5px solid #f40552;
    border-radius: 12px;
    height: 70vh;
    overflow-y: scroll;
`;

export const Content = () => {
    const [pokemonSelectedId, setPokemonSelectedId] = useState<
        string | undefined
    >(undefined);
    const handleClick = (pokemonId: string) => {
        setPokemonSelectedId(pokemonId);
    };

    return (
        <>
            <RowStyled className="row">
                <ColStyled className="col-md-7 ">
                    <PokemonList onCardClick={handleClick} />
                </ColStyled>
                <ColStyled className="col-md-5">
                    <PokemonCard pokemonId={pokemonSelectedId} />
                </ColStyled>
            </RowStyled>
        </>
    );
};
