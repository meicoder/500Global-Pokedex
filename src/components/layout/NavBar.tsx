interface NavBarProps {
    title: string;
}

export const NavBar = ({ title }: NavBarProps) => {
    return (
        <div>
            <nav className="navbar navbar-expand-md navbar-dark bg-dark fixed-top">
                <a className="navbar-brand col-sm-3 col-md-2 mr-0 align-items-center">
                    {title}
                </a>
            </nav>
        </div>
    );
};
