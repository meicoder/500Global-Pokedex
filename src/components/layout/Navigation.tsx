interface NavigationProps {
    nextPage: string | undefined;
    previousPage: string | undefined;
    updateUrl: (url: string) => void;
}

export const Navigation = ({
    nextPage,
    previousPage,
    updateUrl
}: NavigationProps) => {
    return (
        <nav>
            <ul className="pagination justify-content-end">
                {previousPage && (
                    <li className="page-item">
                        <a
                            onClick={() => {
                                updateUrl(previousPage);
                            }}
                            className="page-link"
                            href="#"
                        >
                            Previous
                        </a>
                    </li>
                )}

                {nextPage && (
                    <li className="page-item">
                        <a
                            onClick={() => {
                                updateUrl(nextPage);
                            }}
                            className="page-link"
                            href="#"
                        >
                            Next
                        </a>
                    </li>
                )}
            </ul>
        </nav>
    );
};
