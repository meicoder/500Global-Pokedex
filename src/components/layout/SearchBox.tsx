import styled from 'styled-components';
import { Type } from '../../types/PokemonSchema';

const StyledInput = styled.input`
    border: none;
    padding: 15px;
    width: 250px;
    border: 1px solid #120136;
    margin: 15px;
    display: inline-block;
`;

const StyledSelect = styled.select`
    border: none;
    padding: 15px;
    width: 150px;
    border: 1px solid #120136;
    margin: 15px;
    display: inline-block;
`;

interface SearchBoxProps {
    onInputChange: (inputValue: string) => void;
    types: Type[];
}

export const SearchBox = ({ onInputChange, types }: SearchBoxProps) => {
    const handleTypeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
        console.log(e.target.value);
    };
    return (
        <>
            <StyledInput
                onChange={(e) => {
                    onInputChange(e.target.value);
                }}
                className="search"
                type="search"
                placeholder="Type the pokemon name"
            />
            <StyledSelect name="types" onChange={handleTypeChange}>
                {types.map((type: Type) => {
                    return (
                        <option key={type.name} value={type.name}>
                            {type.name}
                        </option>
                    );
                })}
            </StyledSelect>
        </>
    );
};
