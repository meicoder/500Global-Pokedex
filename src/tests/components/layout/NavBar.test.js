import { shallow } from 'enzyme';
import { NavBar } from '../../../components/layout/NavBar';

describe('Tests NavBar component', () => {
    test('should render the component', () => {
        const wrapper = shallow(<NavBar title="Pokemon" />);
        expect(wrapper).toMatchSnapshot();
    });
});
