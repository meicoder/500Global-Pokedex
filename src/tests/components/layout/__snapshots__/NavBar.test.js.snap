// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests NavBar component should render the component 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top"
  >
    <a
      className="navbar-brand col-sm-3 col-md-2 mr-0 align-items-center"
    >
      Pokemon
    </a>
  </nav>
</div>
`;
